packages/frint-router-react/src/Link.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react'; // eslint-disable-line import/no-extraneous-dependencies
+import PropTypes from 'prop-types'; // eslint-disable-line import/no-extraneous-dependencies
+
+export default class Link extends React.Component {
+  static contextTypes = {
+    app: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    to: PropTypes.string.isRequired,
+    exact: PropTypes.bool,
+    type: PropTypes.string,
+    className: PropTypes.string,
+    activeClassName: PropTypes.string,
+    children: PropTypes.node,
+  };
+
+  constructor(...args) {
+    super(...args);
+
+    this.state = {
+      active: false,
+    };
+
+    this.subscription = null;
+  }
+
+  componentDidMount() {
+    this.considerSubscribingToRouter(this.props);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.considerSubscribingToRouter(nextProps);
+  }
+
+  componentWillUnmount() {
+    this.unsubscribeFromRouter();
+  }
+
+  considerSubscribingToRouter(nextProps) {
+    if (typeof nextProps.activeClassName === 'string') {
+      if (!this.subscription ||
+          this.props.to !== nextProps.to ||
+          this.props.exact !== nextProps.exact) {
+        this.resubscribeToRouter(nextProps.to, nextProps.exact);
+      }
+    }
+  }
+
+  resubscribeToRouter(to, exact) {
+    this.unsubscribeFromRouter();
+
+    this.subscription = this.context.app
+      .get('router')
+      .getMatch$(to, { exact })
+      .subscribe((matched) => {
+        if (!matched) {
+          return this.setState({ active: false });
+        }
+
+        return this.setState({ active: true });
+      });
+  }
+
+  unsubscribeFromRouter() {
+    if (this.subscription) {
+      this.subscription.unsubscribe();
+    }
+  }
+
+  handleClick = (e) => {
+    e.preventDefault();
+
+    // TODO avoid pushing route if it's current URL
+    this.context.app
+      .get('router')
+      .push(this.props.to);
+  };
+
+  render() {
+    const {
+      to,
+      children,
+      className,
+      type,
+      activeClassName,
+    } = this.props;
+
+    const linkProps = {
+      onClick: this.handleClick,
+      className: className || '',
+    };
+
+    if (this.state.active) {
+      linkProps.className += ` ${activeClassName}`;
+    }
+
+    if (typeof type === 'undefined') {
```

Comment:
`type` prop is meant to contain `type` property for a `button` element. if somebody passes a `type` as a prop, then a `<button>` gets rendered, otherwise an `<a>`. I'm not even sure what should happen when somebody passes `null` there and what the developer expects in that case.